src/masoniteorm/models/Model.py

```diff
@@ -711,9 +713,19 @@ def update_or_create(cls, wheres, updates):
         total.update(updates)
         total.update(wheres)
         if not record:
-            return self.create(total, id_key=cls.get_primary_key())
-
-        return self.where(wheres).update(total)
+            # if we don't return fresh, we don't get the primary_key that has been used,
+            # and we can't call it from outside the function lest we get a QueryBuilder.
+            #
+            # Without this we are reduced to performing a DIY update_or_create, e.g.:
+            #     ebay_order = EbayOrder.where({'order_id': d['order_id']}).first()
+            #     if not ebay_order:
+            #         ebay_order = EbayOrder.create(d).fresh()
+            #     else:
+            #         ebay_order.save()
+            return self.create(total, id_key=cls.get_primary_key()).fresh()
+
+        rv = self.where(wheres).update(total)
```

Comment:
This doesn't need to have a variable assigned to it since it isnt being used anywhere. 

self.where(wheres).update(total)

Reply:
agreed

Reply:
Sorry, I'm presently mentally stuck in PHP, and none of it looks right.  Least of all my PR.

In Laravel, we do this:

```php
    // $attributes = wheres, $values = updates
    public function updateOrCreate(array $attributes, array $values = [])
    {
        $instance = $this->firstOrNew($attributes);

        $instance->fill($values)->save();

        return $instance;
    }
    
    public function firstOrNew(array $attributes)
    {
        if (is_null($instance = $this->where($attributes)->first())) {
            $instance = $this->related->newInstance($attributes);
        }

        return $instance;
    }
```
I'm not comfortable with [any version] of the code that updates more than 1 row... though at the time, I only intended to modify the return in cases where the record required creation.  

For full consistancy, I guess something like this... forgive me if the last two lines aren't valid, they're best-guess-pseudocode.

````py
    def update_or_create(cls, wheres, updates):
        self = cls()
        record = self.where(wheres).first()
        total = {}
        total.update(updates)
        total.update(wheres)
        if not record:
            return self.create(total, id_key=cls.get_primary_key()).fresh()

        record.update(total)
        return record
```

Hope I'm making some sense.